### PR TITLE
Ui components/form

### PR DESCRIPTION
**Pull request by sammartinez**
*Description of changes:*
 - Setup base form feature branch, fixed merge conflicts from amplify-ui-components to ui-components/form

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-ui-components/package.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
+{
+  "name": "@aws-amplify/ui-components",
+  "version": "0.0.1",
+  "description": "Stencil Component Starter",
```
Comment (bluezebragames) [inline packages/amplify-ui-components/package.json]: Along those lines, this description should change eventually too

File: packages/amplify-ui-components/src/components/amplify-authenticator/amplify-authenticator.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  };
+
+  private handleSignOut = () => {
+    this.authState = AuthState.LoggedOut;
+    this.userData = {};
+    this.creds = {};
+  };
+
+  render() {
+    const tunnerState = {
```
Comment (bluezebragames) [inline packages/amplify-ui-components/src/components/amplify-authenticator/amplify-authenticator.tsx]: Huh, that's a new one

File: packages/amplify-ui-components/LICENSE [hunk 3]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2018
```
Comment (jordanranz) [inline packages/amplify-ui-components/LICENSE]: Change this to 2019

File: packages/amplify-ui-components/Readme.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+# Stencil Component Starter
```
[the remark this answers is not in the document]
Comment (sammartinez) [reply packages/amplify-ui-components/Readme.md]: Yeah we can do that. I can make a note for later adding this.

File: packages/amplify-ui-components/package.json (hunk 1 above)
Comment (sammartinez) [reply packages/amplify-ui-components/package.json]: Okkkkkk

File: packages/amplify-ui-components/LICENSE (hunk 3 above)
Comment (sammartinez) [reply packages/amplify-ui-components/LICENSE]: Done

**Pull request merged**